File: CSharp/DependencyBreakingTests/E/CheckoutTest.cs
using Org.Codecop.Dependencies.E;
using Moq;
using Xunit;
using Xunit.Abstractions;

namespace Org.Codecop.Dependencies.E.Tests
{
    public class CheckoutTest
    {
        // see https://xunit.net/docs/capturing-output
        private readonly ITestOutputHelper output;
        private readonly Mock<IEmailService> emailServiceMock;

        public CheckoutTest(ITestOutputHelper output)
        {
            this.output = output;
            this.emailServiceMock = new Mock<IEmailService>();
        }

        [Fact]
        public void Test5()
        {
            output.WriteLine("note for tester:");
            output.WriteLine("* Accept Newsletter");
            output.WriteLine("* Do not Accept Terms");

            var polkaDotSocks = new Product("Polka-dot Socks");
            var checkout = new Checkout(polkaDotSocks, emailServiceMock.Object);

            Assert.Throws<OrderCancelledException>(() => checkout.ConfirmOrder());
        }
    }
}

File: Java/src/test/java/org/codecop/dependencies/e/CheckoutTest.java
package org.codecop.dependencies.e;

import static org.mockito.Mockito.when;

import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;

public class CheckoutTest {

    @Rule
    public MockitoRule rule = MockitoJUnit.rule();

    @Mock
    private EmailService emailServiceMock;

    @Mock
    private UserConfirmation newsletter;

    @Mock
    private UserConfirmation terms;

    @Test(expected = OrderCancelledException.class)
    public void test5() {
        when(terms.isAccepted()).thenReturn(false);

        Product polkaDotSocks = new Product("Polka-dot Socks");
        Checkout checkout = new Checkout(polkaDotSocks, emailServiceMock) {
            @Override
            protected UserConfirmation createUserConfirmation(String message) {
                if (message.startsWith("Accept")) {
                    return terms;
                }
                return newsletter;
            }
        };

        checkout.confirmOrder();
    }

}
